### PR TITLE
K8s/Folders: Add all features to k8s endpoints

### PR DESCRIPTION
**Pull request by ryantxu**
This exposes everything currently accessible under `/api/folders/...` in the k8s backed group.

Next up -- a frontend feature toggle to use this endpoint rather than /api/folders

### CODE REVIEW

File: pkg/registry/apis/folders/sub_move.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		return nil, err
+	}
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		target := req.FormValue("parent")
+		if target == "" {
+			responder.Error(fmt.Errorf("missing parent parameter"))
+			return
+		}
+
+		f, err := r.service.Move(ctx, &folder.MoveFolderCommand{
```
Comment (ryantxu) [inline pkg/registry/apis/folders/sub_move.go:55]: I am not sure we need this syntax sugar on top of a regular PATCH updating the folder 🤔 

https://github.com/grafana/grafana/blob/31dd3f8fa887ec46d801f623356cbd6c7b4c437d/pkg/registry/apis/folders/legacy_storage.go#L225

ya, i'll remove it

File: pkg/registry/apis/folders/register.go [hunk 2]
```diff
@@ -120,7 +122,8 @@ func (b *FolderAPIBuilder) GetAPIGroupInfo(
 	storage := map[string]rest.Storage{}
 	storage[resourceInfo.StoragePath()] = legacyStore
 	storage[resourceInfo.StoragePath("parents")] = &subParentsREST{b.folderSvc}
-	storage[resourceInfo.StoragePath("children")] = &subChildrenREST{b.folderSvc}
```
Comment (ryantxu) [inline pkg/registry/apis/folders/register.go:123]: children was the same as List (with a folder specified)

File: pkg/registry/apis/folders/sub_access.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		return nil, err
+	}
+	guardian, err := guardian.NewByFolder(ctx, f, ns.OrgID, user)
+	if err != nil {
+		return nil, err
+	}
+	canView, err := guardian.CanView()
+	if err != nil || !canView {
+		return nil, fmt.Errorf("not allowed to view")
+	}
```
Comment (papagian) [inline pkg/registry/apis/folders/sub_access.go:62]: currently we don't need this given the folder service `Get()` [checks the access to the folder](https://github.com/grafana/grafana/blob/ec5bc7c4ab5396ee68f8851de50957a8f578e783/pkg/services/folder/folderimpl/folder.go#L204)
however, probably in the future we will

**Review by papagian — COMMENTED** [review]
I'm trying to test it but I cannot see the k8s APIs in the swagger editor even though I have set `grafanaAPIServer` and `grafanaAPIServerWithExperimentalAPIs` as suggested in #78091
do you happen to know why?

**Comment by ryantxu** [comment]
@papagian -- I'm running with:
```
app_mode = development 

[feature_toggles]
grafanaAPIServerWithExperimentalAPIs = true
```

And then look here:
http://localhost:3000/swagger?urls.primaryName=folder.grafana.app%2Fv0alpha1
http://localhost:3000/openapi/v3/apis/folder.grafana.app/v0alpha1

image

File: pkg/registry/apis/folders/sub_access.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	if err != nil {
+		return nil, err
+	}
+	user, err := appcontext.User(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// Can view is managed here (and in the Authorizer)
+	f, err := r.service.Get(ctx, &folder.GetFolderQuery{
+		UID:          &name,
```
Comment (papagian) [inline pkg/registry/apis/folders/sub_access.go:49]: without setting the `OrgID` it fails with `folder not found`:
```suggestion
		UID:          &name,
		OrgID:        ns.OrgID,
```

**Review by papagian — CHANGES_REQUESTED** [review]
I have tested and `counts` sub-resource works as expected but for the `access` one we have to set also the `OrgID`

**Review by papagian — APPROVED** [review]
The present changes LGTM

I have noticed an swagger issue with the update endpoint: it does not set the `Content-Type` as a result the request fails:
Screenshot 2024-02-06 at 11 00 04 AM

also, when I try it by setting the `Content-Type` properly it fails with a different error (not sure why):

```shell
% curl -X 'PUT' \
  'http://admin:admin@localhost:3000/apis/folder.grafana.app/v0alpha1/namespaces/default/folders/gen-783b1d42-71bb-4e27-a897-e221175bfdb7' \
  -H 'Content-Type: application/json' \
  -H 'accept: application/json' \
  -H 'Content-Type: */*' \
  -d '{
  "apiVersion": "folder.grafana.app/v0alpha1",
  "kind": "Folder",
  "metadata": {},
  "spec": {
   "title": "updated folder"
  }
}'
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "the name of the object (gen-783b1d42-71bb-4e27-a897-e221175bfdb7 based on URL) was undeterminable: name must be provided",
  "reason": "BadRequest",
  "code": 400
}% 
``` 



</p>
</details>

**Comment by ryantxu** [comment]
Have you tried only sending one content type?  I bet the second one is confusing things
``` 
  -H 'Content-Type: application/json' \
  -H 'Content-Type: */*' \
```

**Pull request merged**